src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -104,6 +113,7 @@ public Task<Session> TryCreateServiceSessionAsync(string serviceName, Solution s
         [Obsolete]
         protected virtual Task<Session> CreateServiceSessionAsync(string serviceName, PinnedRemotableDataScope snapshot, object callbackTarget, CancellationToken cancellationToken)
         {
+            snapshot?.Dispose();
```

Comment:
I would like any sort of approach that doesn't result in random disposes being needed in non-apperant codepaths :)